barretenberg/cpp/src/barretenberg/relations/toy_avm/generic_lookup_relation.hpp

```diff
@@ -0,0 +1,469 @@
+/**
```

Comment:
One potential problem with defining lookups as independent relations is the very large number of inverse polynomials that are going to be created.

This will be expensive

Important question: does the relation algebra ensure that, if a given row is *NOT* performing a lookup, the value of the inverse polynomial is 0? If not, I think this will be needed or committing to 1 inverse polynomial per lookup column will be very painful.

Reply:
I checked, the polynomial should be zero when we are not performing lookups